lucille-core/src/main/java/com/kmwllc/lucille/stage/AddRandomField.java

```diff
@@ -0,0 +1,148 @@
+package com.kmwllc.lucille.stage;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.kmwllc.lucille.core.ConfigUtils;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.util.FileUtils;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * Fetches byte data of a given URL field and places data into a specified field
+ * <br>
+ * Config Parameters -
+ * <br>
+ * inputDataPath (String, Optional) : file path to a text file that stores datapoints to be randomly placed into field,
+ *  defaults to numeric data based on cardinality
+ * fieldName (String, Optional) : Field name of field where data is placed, defaults to "data"
+ * cardinality (int, Optional) : size of the subset of datapoints to be grabbed either from
+ *  given datapath or from random numbers
+ * minNumOfTerms (Integer, Optional) : minimum number of terms to be in the field, defaults to null
+ * maxNumOfTerms (Integer, Optional) : maximum number of terms to be in the field, defaults to null
+ * fieldType (FieldType, Optional) : setting for type of field, default or nested, allows for further settings to be easily added
+ */
+public class AddRandomField extends Stage {
+
+  enum FieldType {
+    DEFAULT, NESTED
+  }
+
+  private final String inputDataPath;
+  private final String fieldName;
+  private final int cardinality;
+  private Integer minNumOfTerms;
+  private Integer maxNumOfTerms;
+  private final FieldType fieldtype;
+  private final List<String> dataArr;
+  private final List<String> uniqueValues;
+
+  public AddRandomField(Config config) throws StageException {
+    super(config, new StageSpec().withOptionalProperties("input_data_path", "field_name", "cardinality", "min_num_of_terms",
+        "max_num_of_terms", "field_type"));
+    this.inputDataPath = ConfigUtils.getOrDefault(config, "input_data_path", null);
+    this.fieldName = ConfigUtils.getOrDefault(config, "field_name", "data");
+    this.minNumOfTerms = ConfigUtils.getOrDefault(config, "min_num_of_terms", null);
+    this.maxNumOfTerms = ConfigUtils.getOrDefault(config, "max_num_of_terms", null);
+    this.fieldtype = FieldType.valueOf(ConfigUtils.getOrDefault(config, "field_type", FieldType.DEFAULT.toString()));
+    this.dataArr = this.inputDataPath != null ? getFileData(this.inputDataPath) : null;
```

Comment:
loading dictionaries should be done in start()

Reply:
Moved to start()